xen/arch/x86/setup.c

```diff
@@ -1171,13 +1171,21 @@ void __init noreturn __start_xen(unsigned long mbi_p)
 
     if ( slaunch_active )
     {
-        /* Prepare for TXT-related code. */
-        map_txt_mem_regions();
+        if ( boot_cpu_data.x86_vendor == X86_VENDOR_INTEL )
+        {
+            /* Prepare for TXT-related code. */
+            map_txt_mem_regions();
+        }
+
         /* Measure SLRT here because it gets used by init_e820(), the rest is
          * measured below by tpm_process_drtm_policy(). */
         tpm_measure_slrt();
-        /* Reserve TXT heap and SINIT. */
-        protect_txt_mem_regions();
+
+        if ( boot_cpu_data.x86_vendor == X86_VENDOR_INTEL )
+        {
+            /* Reserve TXT heap and SINIT. */
+            protect_txt_mem_regions();
+        }
```

Comment:
SLB should also be mapped (in case it isn't located in memory pre-mapped by early Xen) and protected (it contains SLRT).

Reply:
Mapped and protected: https://github.com/TrenchBoot/xen/compare/d99f5ae5f46d04d7044dfdf34a04370fda1b3417..4de56cbbf531cf9efb2efd4baf58f66db71b5ae9#diff-961c78e908063b25617eab27634027bfd5ba6bd6d9253c53a5ac322b94a7c791